driven/storage/adapter.go

```diff
@@ -379,3 +379,86 @@ func (m *database) onDataChanged(changeDoc map[string]interface{}) {
 		m.listener.OnCollectionUpdated(coll.(string), recordMap)
 	}
 }
+
+// Get Survey
+func (sa *Adapter) GetSurvey(id string) (*model.Survey, error) {
+
+	filter := bson.M{"_id": id}
+	var list []model.Survey
+	err := sa.db.surveys.Find(filter, &list, &options.FindOptions{})
+	if err != nil {
+		fmt.Printf("error storage.Adapter.GetSurvey(%s) - %s", id, err)
+		return nil, fmt.Errorf("error storage.Adapter.GetSurvey(%s) - %s", id, err)
+	}
+
+	if len(list) > 0 {
+		entry := list[0]
+		return &entry, nil
+	} else {
+		return nil, nil
+	}
+}
+
+// Create Survey
+func (sa *Adapter) CreateSurvey(survey model.Survey) (*model.Survey, error) {
+
+	_, err := sa.db.surveys.InsertOne(survey)
+	if err != nil {
+		fmt.Printf("error storage.Adapter.CreateSurvey(%s) - %s", survey.ID, err)
+		return nil, fmt.Errorf("error storage.Adapter.CreateSurvey(%s) - %s", survey.ID, err)
+	}
+
+	return &survey, nil
+}
+
+// Update Survey
+func (sa *Adapter) UpdateSurvey(user *model.User, survey model.Survey) (*model.Survey, error) {
+
+	if len(survey.ID) > 0 {
+
+		now := time.Now().UTC()
+		filter := bson.M{"_id": survey.ID, "creator_id": user.Claims.Subject}
+		update := bson.M{"$set": bson.M{
+			"org_id":       survey.OrgID,
+			"app_id":       survey.AppID,
```

Comment:
We should not update `org_id` or `app_id` once the record has been created since these should not change